utils/src/array/mod.rs

```diff
@@ -45,32 +44,8 @@ pub trait Array:
     + Display
     + AsRef<[u8]>
     + Deref<Target = [u8]>
-    + for<'a> TryFrom<&'a [u8], Error = <Self as Array>::Error>
-    + for<'a> TryFrom<&'a Vec<u8>, Error = <Self as Array>::Error>
-    + TryFrom<Vec<u8>, Error = <Self as Array>::Error>
     + FixedSize
```

Comment:
nit: Should we instead do `EncodeFixed` here (seems to imply both `Encode + FixedSize`)?

Reply:
Originally I had it that way, I figured this might be more clear. Maybe I can just add it explicitly to be `Decode + Encode + FixedSize + EncodeFixed`, which is sort of redundant since `EncodeFixed` is blanket implemented, but should be good

Reply:
I actually find `EncodeFixed` more clear than `Encode + FixedSize` but maybe that's just me 🦐